app/src/main/kotlin/com/softteco/template/ui/feature/login/LoginScreen.kt

```diff
@@ -45,29 +56,51 @@ fun LoginScreen(
     viewModel: LoginViewModel = hiltViewModel(),
 ) {
     val state by viewModel.state.collectAsState()
+    val lifecycleOwner = LocalLifecycleOwner.current
     Analytics.logInOpened()
 
-    LaunchedEffect(state.loginState) {
-        if (state.loginState is LoginViewModel.LoginState.Success) {
-            Analytics.logInSuccess()
-            onSuccess()
+    DisposableEffect(lifecycleOwner) {
+        val observer = LifecycleEventObserver { _, event ->
+            if (event == Lifecycle.Event.ON_CREATE) viewModel.onCreate()
```

Comment:
Why do we need to know when the screen is created?

Reply:
Because we need to reset ScreenState to the default value when the screen is opened. In the current implementation, navigation is implemented via ScreenState. When we navigate to the SignUpScreen and back, LoginScreenViewModel is not destroyed yet, and screenState is still Navigate(Screen.SignUp) (the reason why we can't reset ScreenState in the init ViewModel block)